src/common/mfu_daos.c

```diff
@@ -1973,6 +1973,51 @@ static inline void init_hdf5_args(struct hdf5_args *hdf5)
     hdf5->ak = NULL;
 }
 
+static int serialize_kv_rec(struct hdf5_args *hdf5, 
+                            daos_key_t dkey,
+                            daos_handle_t *oh,
+                            uint64_t *dk_index,
+                            char *dkey_val,
+                            mfu_daos_stats_t* stats)
+{
+    void        *buf;
+    int         rc;
+    herr_t      err;
+    hvl_t       *kv_val;
+    daos_size_t size = 0;
+
+    /* get the size of the value */
+    rc = daos_kv_get(*oh, DAOS_TX_NONE, 0, dkey_val, &size, buf, NULL);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to fetch object");
+        goto out;
+    }
+    buf = MFU_CALLOC(1, size);
+    if (buf == NULL) {
+        rc = ENOMEM;
+        goto out;
+    }
```

Comment:
It's fine to leave this check for NULL, but it probably would not trigger if actually out of memory.  The MFU allocate functions will typically abort if they fail to acquire memory, e.g.,

https://github.com/hpc/mpifileutils/blob/55255604d87d581eb52a710c2accad31581136e7/src/common/mfu_util.c#L105